### ISSUE TITLE
tlx headers in deb-package

### ISSUE THREAD

**Issue by fabratu** [tlx/tlx#23]
Hi,

I'm currently working with @angriman and avdgrinten as NetworKit-maintainer (https://github.com/networkit/networkit). This involves the preperation to create binary-packages for debian-based OSes. The plan is to create a lib-, dev- and python-package. For the dev-package there exist a dependency on the tlx headers (not for the lib-package, since the static compiled tlx-library is directly linked inside the NetworKit-runtime-lib during creation).

Is there already some kind of header-package from tlx? If not, would it make sense to either backpack it with the NetworKit-dev package or create an seperate tlx-dev package (due to recommendation/requirements of debian also an tlx-lib package)?

Thanks in advance
Fabian

**Comment by bingmann** [comment]
Hi Fabian, thanks for reaching out.
I think a debian package (both tlx-lib and tlx-dev) are the right way to go.
The algorithms and data structures in tlx are reusable and not found in any other library afaik. I also plan to work hard to never break the interfaces.
What do you think?
You probably need a versioned release tarball?

**Comment by lorenzhs** [comment]
The proper names for such packages would be `libtlx0` (for the dynamic library) and `libtlx-dev` (for the headers and static library), btw. The latter should also depend on the former.

**Comment by fabratu** [comment]
Thanks for the swift reply.

Yes a versioned release tarball would be a good place to start. Based on @lorenzhs naming guidance: Since the plan is to never break the ABI, the major number "always" remains 0, but I guess both minor/release revisions are possible? The full name of the library would then be: `libtlx.so.0.0.0`, with two symlinks: `libtlx.so.0` in the lib-package and `libtlx.so` in the dev-package.
Does the build system of tlx support setting the `soname` and versioning directly? 

Based on your replies, `libtlx0` will then provide a shared runtime-lib. Since NetworKit depends on the dev-package, which installs the lib-package as dependency, it would be cleaner for NetworKit to change from a static to a shared link against tlx.

**Comment by bingmann** [comment]
Okay, I'm not too experienced with Debian packaging.
tlx uses cmake and builds shared libraries when `-DTLX_BUILD_SHARED_LIBS=ON`.
I just added SOVERSION and some install COMPONENTS.
What else is needed? I would then package 0.1.0 when all is added to the cmake.

**Comment by jamesjer** [comment]
I cannot speak to Debian packaging, but I am looking at adding a tlx package to Fedora (to be used by freetdi/gala, which is used by freetdi/tdlib, which sagemath wants).  From my perspective, everything is ready to go.  A versioned tarball would be much appreciated.

**Comment by bingmann** [comment]
Done. I've tagged the first version as v0.5.20191212

**Comment by fabratu** [comment]
Sry for the delay, this "project" was on hold for a while. Thanks also for the release - this should be sufficient for everything. Besides the deb-package, I am in parallel preparing conda and other package manager (brew, spack) release of Networkit. Are there any objections releasing a tlx package also via this channels?

**Comment by bingmann** [comment]
No problem, please package it.
But tell me if you find any compilation warnings or error to fix.

**Comment by fabratu** [comment]
Just as an update on the progress: 
conda integration of tlx is done -> https://github.com/conda-forge/libtlx-feedstock

**Comment by fabratu** [comment]
Integration in homebrew-core is done -> https://github.com/Homebrew/homebrew-core

**Comment by fabratu** [comment]
Integration in spack is done -> https://github.com/spack/spack

**Comment by bingmann** [comment]
Sorry for the delay. I tagged a new version v0.5.20200222

**Comment by fabratu** [comment]
Thank you very much!
